StitchCore-iOS/StitchCore-iOS/Core/Internal/StitchAppClientImpl.swift

```diff
@@ -100,12 +100,12 @@ internal final class StitchAppClientImpl: StitchAppClient {
      * - returns: a service client whose type is determined by the `T` type parameter of the `AnyServiceClientProvider`
      *            passed in the `forProvider` parameter.
      */
-    public func serviceClient<T>(forService serviceClientProvider: AnyServiceClientProvider<T>) -> T {
-        return serviceClientProvider.client(
+    public func serviceClient<T>(forService serviceClientProvider: AnyServiceClientProvider<T>) throws -> T {
```

Comment:
[minor] This isn't going to be fun to use if this can throw. Can we avoid this somehow?